lib/ci-stack.ts

```diff
@@ -80,7 +85,13 @@ export class CIStack extends Stack {
       oidcCredArn: importedOidcConfigValuesSecretBucketValue.toString(),
       useSsl,
       runWithOidc,
-    });
+    },
+    {
+      agentNodeSecurityGroup: securityGroups.agentNodeSG.securityGroupId.toString(),
```

Comment:
Don't toString these, you can pass around the security group and have it tostring before its used if needed